### PR TITLE
[v7r2] add histogram support to DIRAC, add integration test and make Graph package work with python3

### PR DESCRIPTION
**Pull request by zmathe**
BEGINRELEASENOTES
*Core
NEW: add histogram support to the Graph package.
NEW: add integration tests
NEW: adopt Graph package to python3
ENDRELEASENOTES

### CODE REVIEW

File: Core/Utilities/Graphs/GraphData.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def setData(self, data):
 
   def initialize(self, key_type=None):
 
-    keys = self.data.keys()
+    keys = list(self.data.keys())
```
Comment (fstagni) [inline Core/Utilities/Graphs/GraphData.py:88]:
```suggestion
    keys = list(self.data)
```

File: Core/Utilities/Graphs/GraphData.py [hunk 2]
```diff
@@ -146,13 +146,13 @@ def sortLabels(self, sort_type='max_value', reverse_order=False):
         self.label_values = [self.plotdata.parsed_data[l] for l in self.labels]
     else:
       if sort_type == 'max_value':
-        pairs = zip(self.subplots.keys(), self.subplots.values())
+        pairs = list(zip(self.subplots.keys(), self.subplots.values()))
```
Comment (fstagni) [inline Core/Utilities/Graphs/GraphData.py:149]: This does not look correct. Did you mean
```suggestion
        pairs = list(zip(list(self.subplots), self.subplots.values())))
```
?

File: Core/Utilities/Graphs/GraphData.py [hunk 3]
```diff
@@ -146,13 +146,13 @@ def sortLabels(self, sort_type='max_value', reverse_order=False):
...
         reverse = not reverse_order
         pairs.sort(key=lambda x: x[1].max_value, reverse=reverse)
         self.labels = [x[0] for x in pairs]
         self.label_values = [x[1].max_value for x in pairs]
       elif sort_type == 'last_value':
-        pairs = zip(self.subplots.keys(), self.subplots.values())
+        pairs = list(zip(self.subplots.keys(), self.subplots.values()))
```
Comment (fstagni) [inline Core/Utilities/Graphs/GraphData.py:155]: Same here.

File: Core/Utilities/Graphs/GraphData.py (hunk 1 above)
Comment (fstagni) [reply Core/Utilities/Graphs/GraphData.py:88]: You have similar changes below.

File: Core/Utilities/Graphs/Legend.py [hunk 4]
```diff
@@ -181,8 +178,8 @@ def draw(self):
       else:
         num = None
       color = self.palette.getColor(label)
-      row = nc % nRows
-      column = nc / nRows
+      row = int(nc % nRows)
```
Comment (fstagni) [inline Core/Utilities/Graphs/Legend.py:181]: Here `int` is not needed

File: tests/Integration/AccountingSystem/Test_Plots.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
+###############################################################################
+# (c) Copyright 2019 CERN for the benefit of the LHCb Collaboration           #
+#                                                                             #
+# This software is distributed under the terms of the GNU General Public      #
+# Licence version 3 (GPL Version 3), copied verbatim in the file "LICENSE".   #
+#                                                                             #
+# In applying this licence, CERN does not waive the privileges and immunities #
+# granted to it by virtue of its status as an Intergovernmental Organization  #
+# or submit itself to any jurisdiction.                                       #
+###############################################################################
```
Comment (fstagni) [inline tests/Integration/AccountingSystem/Test_Plots.py:10]: This should not be here.

Comment (zmathe) [reply tests/Integration/AccountingSystem/Test_Plots.py:10]: What do you mean? This file? or the copyright?

File: Core/Utilities/Graphs/Legend.py (hunk 4 above)
Comment (zmathe) [reply Core/Utilities/Graphs/Legend.py:181]: it is needed for python3

File: tests/Integration/AccountingSystem/Test_Plots.py (hunk 5 above)
Comment (fstagni) [reply tests/Integration/AccountingSystem/Test_Plots.py:10]: The copyright.

File: Core/Utilities/Graphs/Legend.py (hunk 4 above)
Comment (fstagni) [reply Core/Utilities/Graphs/Legend.py:181]: The `%` operator will give always an `int` in output.

Comment (zmathe) [reply Core/Utilities/Graphs/Legend.py:181]: Federico, I wrote integration tests in python2 and after I executed the tests in python3 and I spent lot of times to fix the bugs and now you tell it is always int. 
python3:
```
>>> 3/2
1.5
>>> 
```
python2
```
>>> 3/2
1
>>> 
```

Comment (fstagni) [reply Core/Utilities/Graphs/Legend.py:181]: yes but in this line (181) there's no division: there's the modulo operator.

Comment (zmathe) [reply Core/Utilities/Graphs/Legend.py]: sorry. the line above was the problematic one.

**Comment by zmathe** [comment]
all applied all your corrections

File: Core/Utilities/Graphs/BarGraph.py [hunk 6]
```diff
@@ -144,7 +144,7 @@ def draw(self):
     if 'log_yaxis' in self.prefs:
       ymin = 0.001
     else:
-      ymin = min(tmp_b, 0.)
+      ymin = min(min(tmp_b), 0.)
```
Comment (fstagni) [inline Core/Utilities/Graphs/BarGraph.py:147]: Do you need this?

**Pull request merged**